File: src/main/java/de/silencio/activecraftcore/utils/config/LocationsConfig.java
package de.silencio.activecraftcore.utils.config;

import lombok.Getter;
import org.bukkit.Location;

import java.util.HashMap;

@Getter
public class LocationsConfig extends ActiveCraftConfig {

    private HashMap<String, Location> locations;
    private Location spawn;

    public LocationsConfig() {
        super(new FileConfig("locations.yml"));
    }

    @Override
    protected void load() {
        locations = new HashMap<>();
        fileConfig.getKeys(false).stream()
                .filter(key -> fileConfig.getLocation(key) != null)
                .forEach(key -> locations.put(key, fileConfig.getLocation(key)));
        spawn = locations.get("spawn");
    }
}

File: src/main/java/org/activecraft/activecraftcore/utils/config/LocationsConfig.kt
package org.activecraft.activecraftcore.utils.config

import org.bukkit.Location

class LocationsConfig : ActiveCraftConfig("locations.yml") {
    lateinit var locations: Map<String, Location>
        private set
    var spawn: Location? = null

    override fun load() {
        locations = fileConfig.getKeys(false)
            .filter { fileConfig.getLocation(it) != null }
            .associateWith { fileConfig.getLocation(it)!! }
        spawn = locations["spawn"]
    }
}
